scripts/opencl/opencl.go

```diff
@@ -36,28 +36,28 @@ func compileSources() (string, error) {
 	dir := filepath.Join(filepath.Dir(filename), sourceDir)
 	files, err := ioutil.ReadDir(dir)
 	if err != nil {
-		return "", errors.Errorf("couldn't read directory %s: %s", dir, err.Error())
+		return "", errors.Wrapf(err, "read directory: %s", dir)
 	}
 
 	contents := make(map[string][]byte)
 	for _, file := range files {
 		filepath := filepath.Join(dir, file.Name())
 		data, err := ioutil.ReadFile(filepath)
 		if err != nil {
-			return "", errors.Errorf("couldn't read file %s: %s", filepath, err.Error())
+			return "", errors.Wrapf(err, "read file: %s", filepath)
 		}
 		contents[file.Name()] = data
 	}
 	var out strings.Builder
 	if _, ok := contents[headerFile]; !ok {
-		return "", errors.Errorf("missing %s file", headerFile)
+		return "", errors.Wrapf(err, "missing %s file", headerFile)
 	}
 	out.Write(contents[headerFile])
 	delete(contents, headerFile)
 
 	mainSrc, ok := contents[kernelFile]
 	if !ok {
-		return "", errors.Errorf("missing %s file", kernelFile)
+		return "", errors.Wrapf(err, "missing %s file", kernelFile)
```

Comment:
Where is this error coming from? Please double-check the entire PR for such issues.

Reply:
When I didn't have a declared error variable I put `errors.Errorf(msg)`
I will change it to `Wrapf` it is better to have control of errors.

Reply:
I put missing kernel and missing header

Reply:
The question still stands, where does this `err` variable come from?